ts/Stock/Indicators/DisparityIndex/DisparityIndexIndicator.ts

```diff
@@ -0,0 +1,282 @@
+    public options: DisparityIndexOptions = void 0 as any;
+    public points: Array<DisparityIndexPoint> = void 0 as any;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    public init(this: DisparityIndexIndicator): void {
+        const args = arguments,
+            ctx = this, // Disparity Index indicator
+            params = args[1].params, // options.params
+            averageType = params && params.average ? params.average : void 0;
+
+        let averageIndicator;
+
+        switch (averageType) {
+        case 'ema':
+            averageIndicator = EMAIndicator;
+            break;
+        case 'dema':
+            averageIndicator = DEMAIndicator;
+            break;
+        case 'tema':
+            averageIndicator = TEMAIndicator;
+            break;
+        case 'wma':
+            averageIndicator = WMAIndicator;
+            break;
+        default: // use sma if any of the above strings do not match
+            averageIndicator = SMAIndicator;
+        }
+
+        ctx.averageIndicator = averageIndicator;
+
+        if (averageType !== 'sma') { // no need to check whether sma is loaded
+            // Check if the required average indicator modules is loaded
+            RequiredIndicatorMixin.isParentLoaded(
+                averageIndicator as any,
+                averageType,
+                ctx.type,
+                function (indicator: Highcharts.Indicator): undefined {
+                    indicator.prototype.init.apply(ctx, args);
+                    return;
+                }
+            );
+        }
+    }
+
+    public calculateDisparityIndex(
+        close: number,
+        periodAverage: number
+    ): number {
+        return correctFloat(close - periodAverage) / periodAverage * 100;
+    }
+
+    public getValues<TLinkedSeries extends LineSeries>(
+        series: TLinkedSeries,
+        params: DisparityIndexParamsOptions
+    ): (IndicatorValuesObject<TLinkedSeries>|undefined) {
+        const averageType = params.average,
+            index = params.index,
+            period = params.period as any,
+            xVal: Array<number> = (series.xData as any),
+            yVal: Array<number>|Array<Array<number>> = (series.yData as any),
+            yValLen: number = yVal ? yVal.length : 0,
+            disparityIndexPoint: Array<Array<number>> = [],
+            xData: Array<number> = [],
+            yData: Array<number> = [],
+            averageIndicator: any = this.averageIndicator,
+            isOHLC = isArray(yVal[0]);
+
+        // average indicator period minus 1
+        let range;
+
+        switch (averageType) {
+        case 'dema':
+            range = (2 * period - 1) - 1;
+            break;
+        case 'tema':
+            range = (3 * period - 2) - 1;
+            break;
+        case 'ema':
+        case 'wma':
+        default: // use sma if any of the above strings do not match
+            range = period - 1;
+        }
```

Comment:
Do we need this checking? Instead of manually checking the period, can we base on what's returned from `getValues()`? If `getValues()` returns `undefined`, then we don't have enough points => return undefined;

We can get the first point from: 

```js
const start = xVal.indexOf(getValues().xData[0]);
for (let i = start; i < yValLen ; i++) { ... }
```

Reply:
Nice idea. I'll check it